tracer/src/Datadog.Trace/IAST/IastRequestContext.cs

```diff
@@ -0,0 +1,44 @@
+// <copyright file="IastRequestContext.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+
+namespace Datadog.Trace.Iast;
+
+internal class IastRequestContext
+{
+    private VulnerabilityBatch? vulnerabilityBatch;
+
+    internal void AddIastTagsIfNeeded(Span span)
+    {
+        if (Iast.Instance.Settings.Enabled && span.IsRootSpan && span.Type == SpanTypes.Web)
+        {
+            AddIastInfoToRootSpan(span);
+        }
+    }
+
+    private void AddIastInfoToRootSpan(Span span)
+    {
+        // Right now, we always set the IastEnabled tag to "1", but in the future, it will not be added if iast is enabled but the request is not analyzed.
+        span.Tags.SetTag(Tags.IastEnabled, "1");
+
+        if (vulnerabilityBatch != null)
+        {
+            span.Tags.SetTag(Tags.IastJson, vulnerabilityBatch.ToString());
+        }
+    }
+
+    internal void AddVulnerability(Vulnerability vulnerability)
+    {
+        if (vulnerabilityBatch == null)
+        {
+            vulnerabilityBatch = new();
```

Comment:
Could there be a race condition here if accessed from multiple threads?

Reply:
I thought that even if IastRequestContext is initializaed, we might not find any vulnerability (which will happen most of the time), so making vulnerabilitybatch lazy would improve the performance.